File: CookUs/src/api/faq.ts
import api from './axios'

export type FaqItem = {
  faq_id: number
  question: string
  answer: string
  category?: string | null
  created_at: string
  updated_at?: string
  is_visible?: number
}

export async function fetchFaq(query = '', category?: string, limit = 30): Promise<FaqItem[]> {
  const { data } = await api.get('/faq', { params: { query: query || undefined, category: category || undefined, limit } })
  return (data?.items ?? []) as FaqItem[]
}

export async function fetchFaqCategories(): Promise<string[]> {
  const { data } = await api.get('/faq/categories')
  return (data?.items ?? []) as string[]
}

File: Backend/app/schemas/faq.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class FaqItem(BaseModel):
    faq_id: int
    question: str
    answer: str
    category: Optional[str] = None
    created_at: Optional[datetime] = None


__all__ = ["FaqItem"]
